test/tools/ossfuzz/SolidityMutator.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/tools/ossfuzz/SolidityMutator.h>
+
+#include <libsolutil/Whiskers.h>
+
+#include <boost/algorithm/string/join.hpp>
+
+#include <algorithm>
+
+using namespace std;
+using namespace solidity::util;
+using namespace solidity::test::fuzzer;
+
+string SolidityMutator::genRandString(unsigned int _length, char _start, char _end)
+{
+	uniform_int_distribution<char> dist(_start, _end);
+	string result{};
+	generate_n(back_inserter(result), _length, [&]{ return dist(Rand); });
+	return result;
+}
+
+antlrcpp::Any SolidityMutator::visitPragmaDirective(SolidityParser::PragmaDirectiveContext*)
+{
+	auto PickLiteral = [](unsigned const _idx) -> string
+	{
+		static const vector<string> alphanum = {
+			"0",
+			"experimental",
+			"solidity < 142857",
+			"solidity >=0.0.0",
+			"solidity >=0.0.0 <0.8.0",
+			"experimental __test",
+			"experimental SMTChecker",
+			"experimental ABIEncoderV2",
+			"experimental \"xyz\""
+		};
+
+		return alphanum[_idx % alphanum.size()];
+	};
+
+	string pragma{};
+	if (coinFlip())
+		pragma = PickLiteral(Rand());
+	else
+	{
+		// Generate a random pragma that does not contain a semicolon
+		pragma = genRandString(randOneToN(30), '<');
+		pragma.erase(remove(pragma.begin(), pragma.end(), ';'), pragma.end());
+	}
+
+	Out << Whiskers(R"(pragma <string>;<nl>)")
+		("string", pragma)
+		("nl", "\n")
+		.render();
+	return antlrcpp::Any();
```

Comment:
Does it actually work? I see that the mutated code is returned via `Out` but it's a member variable and not something inherited.

Is this class is just used as a template parameter by something in libfuzzer? That would explain the weird naming convention...

Reply:
Sorry for the naming convention. No, it is not due to something in the libfuzzer. I suppose instead of a stringstream member variable, I could also have the visitors return a string and keep concatenating but I think that is more of a cosmetic change.

Reply:
But what is accessing the stringstream? `SolidityMutator` is a newly added class and it's not a specialization of an existing class. There's nothing in this PR that gets the value out of `Out`.

Reply:
Oh, I see it now. The mutator has `toString()` method and you do call it. Not sure how I missed that.